django/db/backends/base/schema.py

```diff
@@ -798,7 +799,9 @@ def _create_index_name(self, model, column_names, suffix=""):
             )
         # Else generate the name for the index using a different algorithm
         table_name = model._meta.db_table.replace('"', '').replace('.', '_')
-        index_unique_name = '_%x' % abs(hash((table_name, ','.join(column_names))))
```

Comment:
What are the backwards compatibility concerns with changing the algorithm? I was thinking maybe we should keep the current method on Python 2 and only have a new one for Python 3. On the other hand, then there will be upgrade issues when going from Python 2 to 3.